### PR TITLE
Remove conmon cgroup before pod cgroup for cgroupfs

### PR DESCRIPTION
**Pull request by mheon**
For pods using cgroupfs, we were seeing some error messages in CI from an inability to remove the pod CGroup, which was traced down to the conmon cgroup still being present as a child. Try to
remove these error messages and ensure successful CGroup deletion by removing the conmon CGroup first, then the pod cgroup.

### CODE REVIEW

File: libpod/runtime_pod_linux.go [hunk 1]
```diff
@@ -265,7 +265,19 @@ func (r *Runtime) removePod(ctx context.Context, p *Pod, removeCtrs, force bool)
 			}
 		case CgroupfsCgroupsManager:
 			// Delete the cgroupfs cgroup
+			// Make sure the conmon cgroup is deleted first
 			v1CGroups := GetV1CGroups(getExcludedCGroups())
+			conmonCgroup, err := cgroups.Load(v1CGroups, cgroups.StaticPath(filepath.Join(p.state.CgroupPath, "conmon")))
+			if err != nil && err != cgroups.ErrCgroupDeleted {
+				return err
+			} else if err == nil {
```
Comment (rhatdan) [inline libpod/runtime_pod_linux.go]: No need for else

**Comment by TomSweeneyRedHat** [comment]
@mheon tests aren't too hip.  Other than @rhatdan's comment, LGTM.

**Comment by mheon** [comment]
Rebased, let's see if CI is fixed

**Comment by mheon** [comment]
@rhatdan @baude @TomSweeneyRedHat @umohnani8 Anyone mind dropping an LGTM?

File: libpod/runtime_pod_linux.go (hunk 1 above)
Comment (mheon) [reply libpod/runtime_pod_linux.go]: Fixed

**Comment by umohnani8** [comment]
LGTM

**Pull request merged**